packages/addons-inject/tests/wrangler.test.js

```diff
@@ -0,0 +1,256 @@
+      `<script src="https://assets.readthedocs.org/static/core/js/readthedocs-doc-embed.js"></script>`,
+    ];
+    const links = [
+      `<link rel="stylesheet" href="/_/static/css/readthedocs-doc-embed.css" />`,
+      `<link rel="stylesheet" href="https://assets.readthedocs.org/static/css/readthedocs-doc-embed.css" />`,
+      `<link rel="stylesheet" href="https://assets.readthedocs.org/static/css/badge_only.css" />`,
+      `<link rel="stylesheet" href="/_/static/css/badge_only.css" />`,
+    ];
+    const elements = [
+      `<div role="main"><div x-ref="outer"><div class="admonition warning" x-ref="inner"></div></div></div>`,
+      `<div role="main" x-ref="outer-furo"><div class="admonition warning" x-ref="inner-furo"></div></div>`,
+      `<div id="main-content"><div><div><article x-ref="outer-book"><div class="admonition warning" x-ref="inner-book"></div></article></div></div></div>`,
+      `<div class="rst-versions" x-ref="flyout"></div>`,
+    ];
+
+    // Ensure we're testing all of the removals
+    expect(scripts.length).toBe(AddonsConstants.removalScripts.length);
+    expect(links.length).toBe(AddonsConstants.removalLinks.length);
+    expect(elements.length).toBe(AddonsConstants.removalElements.length);
+
+    // Test all removals at once
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/" })
+      .reply(
+        200,
+        `
+        <html>
+          <head>
+            ${scripts.join("")}
+            ${links.join("")}
+            <script src="https://example.com"></script>
+          </head>
+          <body>
+            ${elements.join("")}
+          </body>
+        </html>`,
+        {
+          headers: {
+            "Content-type": "text/html",
+            "X-RTD-Force-Addons": true,
+            "X-RTD-Hosting-Integrations": false,
+          },
+        },
+      );
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/",
+    );
+    expect(response.status).toBe(200);
+    const matcher = expect(await response.text());
+    matcher.to.contain(AddonsConstants.scriptAddons);
+    // Ensure matches aren't too aggressive
+    matcher.to.contain(`<script src="https://example.com"></script>`);
+
+    // Make sure scripts and linker were removed
+    for (const script of scripts) {
+      matcher.to.not.contain(script);
+    }
+    for (const link of links) {
+      matcher.to.not.contain(link);
+    }
+
+    // Make sure elements were removed, using extra attrs because the whole
+    // string is not removed
+    matcher.to.contain(`x-ref="outer"`);
+    matcher.to.not.contain(`x-ref="inner"`);
+    matcher.to.contain(`x-ref="outer-furo"`);
+    matcher.to.not.contain(`x-ref="inner-furo"`);
+    matcher.to.contain(`x-ref="outer-book"`);
+    matcher.to.not.contain(`x-ref="inner-book"`);
+    matcher.to.not.contain(`x-ref="flyout"`);
```

Comment:
I'm not sure to understand this even with the context in the comment. Can you expand on why we are checking this extra attrs? 😄

Reply:
When removing `<script>` elements, the change removes the entire element. From:

```
<head><script src="..."></script></head>
```

When removing HTML elements, we're only targeting the child elements for removal:

That is, a source:

```
<body><div><div><div class="rst-versions"></div></div></div></body>
```

After processing and removing the child flyout element, the parent `<div>` elements still exist:

```
<body><div><div></div></div></body>
```

The attributes help test for elements explicitly instead of guessing how each remaining `<div><div></div></div>` aligns with each source example.